src/ext/background.js

```diff
@@ -207,10 +209,45 @@ function processHeaders(details, url) {
  */
 function beforeSendHeaders(request, url) {
     // Cancel if we don't have a token to spend
-    if (!DO_REDEEM || !getSpendFlag(url.host) || checkMaxSpend(url.host) || spentUrl[url.href] || isErrorPage(url.href) || isFaviconUrl(url.href) || REDEEM_METHOD != "reload") {
+
+    let reqUrl = url.href;
+
+    // No reload method branch
+    if (DO_REDEEM && REDEEM_METHOD === "no-reload" && !isErrorPage(reqUrl) && !isFaviconUrl(reqUrl)) {
+        // check that we're at an URL that can handle redeems
+        const isRedeemUrl = ACTIVE_CONFIG["spend-action"]["urls"]
+            .map(redeemUrl => patternToRegExp(redeemUrl))
+            .filter(re => reqUrl.match(re)).length > 0;
+
+        if (countStoredTokens() > 0 && isRedeemUrl) {
+
+            const tokenToSpend = GetTokenForSpend();
+            if (tokenToSpend == null) {
+                return {cancel: false};
+            }
+            setSpendFlag(url.host, null);
+            incrementSpentHost(url.host);
+
+            const http_path = request.method + " " + url.pathname;
+            const redemptionString = BuildRedeemHeader(tokenToSpend, url.hostname, http_path);
+            let headers = request.requestHeaders
+            headers.push({name: ACTIVE_CONFIG["spend-action"]["header-name"], value: redemptionString});
+            headers.push({name: ACTIVE_CONFIG["spend-action"]["header-host-name"], value: url.hostname});
+            headers.push({name: ACTIVE_CONFIG["spend-action"]["header-path-name"], value: http_path});
+            spendId[request.requestId] = true;
+            spentUrl[reqUrl] = true;
+            return {requestHeaders: headers};
+        }
+    }
+
+    if (!DO_REDEEM || !getSpendFlag(url.host) || checkMaxSpend(url.host) || spentUrl[reqUrl] || isErrorPage(reqUrl) || isFaviconUrl(url.href) || REDEEM_METHOD !== "reload") {
+        // TODO reverse this branch, end up with returning {cancel: false}
```

Comment:
Agree, we should return {cancel:false} at the end of the function.

Reply:
Done, could use some 👀